### PR TITLE
feat(snowflake-usage): Generate email address if not exists 

### PR DESCRIPTION
**Pull request by treff7es**
This will introduce a breaking change as users will have to specify the `email_domain` config parameter.

## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/usage/snowflake_usage.py [hunk 1]
```diff
@@ -335,6 +336,14 @@ def is_object_valid(obj: Dict[str, Any]) -> bool:
                 event_dict["query_start_time"]
             ).astimezone(tz=timezone.utc)
 
+            if not event_dict["email"] and self.config.email_domain:
+                if not event_dict["user_name"]:
+                    event_dict["user_name"] = "unknown"
```
Comment (jjoyce0510) [inline metadata-ingestion/src/datahub/ingestion/source/usage/snowflake_usage.py:341]: If we don't have a username we are falling back to unknown? This seems kinda strange - why not just use None?

Comment (treff7es) [reply metadata-ingestion/src/datahub/ingestion/source/usage/snowflake_usage.py]: I changed it to skip users without email address or username

File: metadata-ingestion/src/datahub/ingestion/source/usage/snowflake_usage.py [hunk 2]
```diff
@@ -335,6 +336,17 @@ def is_object_valid(obj: Dict[str, Any]) -> bool:
                 event_dict["query_start_time"]
             ).astimezone(tz=timezone.utc)
 
+            if not event_dict["email"] and self.config.email_domain:
```
Comment (jjoyce0510) [inline metadata-ingestion/src/datahub/ingestion/source/usage/snowflake_usage.py:339]: Looks great!

**Review by jjoyce0510 — APPROVED** [review]
LGTM

**Review by shirshanka — APPROVED** [review]
LGTM

**Pull request merged**